web/src/compose_notifications.ts

```diff
@@ -298,6 +298,21 @@ export function notify_local_mixes(
             continue;
         }
 
+        if (
+            !message_lists.current?.data.fetch_status.has_found_newest &&
+            narrow_state.filter()?.is_conversation_view_with_near()
+        ) {
+            // For `near` narrows, don't force user to jump to the sent message.
+            // This fixes a bug where trying to jump user to `link_msg_id` via `then_select_id`
+            // would fail because there are older unreads in the narrow and the `narrow_to_recipient`
+            // moves user out of the `near` narrow. This makes us jump to the first unread if present
+            // instead of the sent message.
+            // TODO: Refactor `notify_above_composebox` to keep user in the `near` narrow
+            // on clicking the compose banner link.
+            show_scroll_to_view_banner(link_msg_id);
```

Comment:
Hmm, but is this correct if `need_user_to_scroll` is false? Like what if you were already at the bottom and the messages is immediately visible?

Reply:
In that case, we will never reach here since `message_lists.current?.data.fetch_status.has_found_newest()` would be true.